pkg/controller/controller_test.go

```diff
@@ -2193,3 +1770,218 @@ func getRecordedEvents(source <-chan string) []string {
 	}
 	return events
 }
+
+func assertNumEvents(t *testing.T, strings []string, number int) {
+	if e, a := number, len(strings); e != a {
+		t.Fatalf("Unexpected number of events: expected %v, got %v", e, a)
```

Comment:
I'd rather do it in a follow-up, if that's okay with you.  It takes quite a bit of time to comb through this file, and I'd rather get it in as-is than go up to the line on the release.  Thoughts?